src/octopeer-github/main/Settings/Settings.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * Created by Mitchell on 12-5-2016.
+ */
+
+/**
+ * Class from which user settings can be retrieved.
+ * These settings have been maintained by the front-end.
+ */
+class Settings {
+    private logging: Boolean;
+    private tabs: Boolean;
+    private comments: Boolean;
+    private peer_comments: Boolean;
+    private focus: Boolean;
+    private username: Boolean;
+    private repo: Boolean;
+    private file: Boolean;
+
+    /**
+     * Gets the logging preference, from the chrome storage.
+     * @returns {Boolean}
+     */
+    public getLogging() {
+        chrome.storage.sync.get("loggingEnabled", function (obj) {
+            this.logging = obj;
+        });
+        return this.logging;
```

Comment:
Does this work? As a callback usually indicates that the call is asynchronous and thus is not executed immediately, meaning `this.logging` is only set after this method has returned